File: selfstudy/python/students_filterlow.py
'''
task 116:
Students - FilterLow
You need to return only the values less than 50 from student test scores
'''

def below_fifty_students(scores):
    return [x for x in scores if x < 50]

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], below_fifty_students([10, 20, 30])),
        ([50, 60, 70], below_fifty_students([50, 60, 70])),
        ([1, 2, 3], below_fifty_students([1, 2, 3])),
        ([100, 200], below_fifty_students([100, 200])),
        ([], below_fifty_students([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = below_fifty_students(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# finally works

File: selfstudy/golang/students_filterlow.go
/*
task 216:
Students - FilterLow
You need to return only the values less than 50 from student test scores
*/

package main

import (
	"fmt"
	"reflect"
)

func Below_fiftyStudents(scores []int) []int {
	var res []int
	for _, v := range scores {
		if v < 50 { res = append(res, v) }
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Below_fiftyStudents([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Below_fiftyStudents([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Below_fiftyStudents([]int{1, 2, 3})},
		{[]int{100, 200}, Below_fiftyStudents([]int{100, 200})},
		{[]int{}, Below_fiftyStudents([]int{})},
	}
	
	for idx, t := range tests {
		res := Below_fiftyStudents(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think
